src/stream/src/executor/over_window/eowc.rs

```diff
@@ -661,26 +622,19 @@ mod tests {
             frame: Frame::rows(FrameBound::Preceding(1), FrameBound::Following(1)),
         }];
 
-        let (mut tx, mut over_window) = create_executor(calls.clone(), store.clone()).await;
-
-        tx.push_barrier(1, false);
-        over_window.expect_barrier().await;
-
-        tx.push_chunk(StreamChunk::from_pretty(
-            " I T  I   i
-            + 1 p1 100 10
-            + 1 p1 101 16
-            + 4 p1 102 20",
-        ));
-        assert_eq!(1, over_window.expect_watermark().await.val.into_int64());
-        let chunk = over_window.expect_chunk().await;
-        assert_eq!(
-            chunk,
-            StreamChunk::from_pretty(
-                " I T  I   i  I
-                + 1 p1 100 10 26
-                + 1 p1 101 16 46"
+        insta::assert_snapshot!(
+            executor_snapshot(
+                Box::new(async || create_executor(calls.clone(), store.clone()).await),
```

Comment:
Is async closure inevitable here?

Reply:
I don't know (yet). Hope a professional to tell me. 🥵

Reply:
What about a closure returning a `Future`? 🥵

Reply:
Oh damn